tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs

```diff
@@ -0,0 +1,398 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Test.Common;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Integration.Infrastructure;
+using Microsoft.Identity.Test.Integration.net45.Infrastructure;
+using Microsoft.Identity.Test.Integration.NetFx.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+
+    [TestClass]
+    public class ClientCredentialsTests
+    {
+        private static readonly string[] s_scopes = { "User.Read" };
+        private static readonly string[] s_keyvaultScope = { "https://vault.azure.net/.default" };        
+        private const string PublicCloudConfidentialClientID = "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8";
+        private const string PublicCloudTestAuthority = "https://login.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47";
+
+
+        private enum CredentialType { 
+            Cert, 
+            Secret, 
+            ClientAssertion_Wilson, 
+            ClientAssertion_Manual,
+            ClientClaims_MergeClaims,
+            ClientClaims_ExtraClaims
+        };
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            TestCommon.ResetInternalStaticCaches();          
+        }
+    
+        [TestMethod]
+        [DataRow(Cloud.Public, RunOn.NetFx | RunOn.NetCore)]
+        [DataRow(Cloud.Adfs, RunOn.NetCore)]
+        [DataRow(Cloud.PPE, RunOn.NetFx)]
```

Comment:
This is how I see integration tests being organized. We could move the data rows to be config files, but I think this is reasonably easy to read. 

I also disabled some combinations of cloud x framework because it is unlikely to ever fail. This is called `pairwise testing`. If the number of dimensions grows, we can use a tool like https://github.com/microsoft/pict to more formally generate a reasonable number of tests.